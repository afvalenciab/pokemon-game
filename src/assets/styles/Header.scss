.home--title {
  grid-area: title;

  figure{
    margin: 0;
    img {
      vertical-align: top;
      width: 300px;
    }
  }
}