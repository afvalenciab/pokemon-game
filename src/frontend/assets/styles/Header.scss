.home--title {
  grid-area: title;

  figure{
    margin: 0;
    img {
      vertical-align: top;
      width: 300px;
    }
  }
}

@media screen and (max-width: 650px) {
  .home--title {
    figure{
      img {
        width: 150px;
      }
    }
  }
}