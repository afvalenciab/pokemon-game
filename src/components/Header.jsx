import React from 'react';
import logo from '../assets/static/Pokemon_logo.svg';
import '../assets/styles/Header.scss';

const Header = () => {
  return (
    <header className='home--title'>
      <figure>
        <img src={logo} alt='Pokemon Logo' />
      </figure>
    </header>
  );
};

export default Header;
