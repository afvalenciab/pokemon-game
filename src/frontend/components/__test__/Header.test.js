import React from 'react';
import { create } from 'react-test-renderer';
import { render,shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Header from '../Header';

configure({ adapter: new Adapter() });

describe('Header Testing', () => {
  test('Match Snapshot', () => {
    const header = create(<Header />);
    expect(header.toJSON()).toMatchSnapshot();
  });

  test('Header has .home--title class', () => {
    const header = shallow(<Header />);
    const headerElem = header.find('header');
    expect(headerElem.hasClass('home--title')).toBe(true);
  });

  test('header has 1 img tag', () => {
    const header = render(<Header />);
    expect(header.find('img')).toHaveLength(1);
  });
});