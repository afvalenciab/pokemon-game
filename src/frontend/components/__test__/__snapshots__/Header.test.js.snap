// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Testing Match Snapshot 1`] = `
<header
  className="home--title"
>
  <figure>
    <img
      alt="Pokemon Logo"
      src="test-file-stubs"
    />
  </figure>
</header>
`;
